backend/buildeR.Common/Services/EmailService.cs

```diff
@@ -16,13 +17,29 @@ public class EmailService : IEmailService
         private readonly string _apiKey;
         private readonly IEmailBuilder _builder;
         private readonly SendGridClient _client;
-        public EmailService(IEmailBuilder builder)
+        private readonly IConfiguration _configuration;
+        public EmailService(IEmailBuilder builder, IConfiguration configuration)
         {
             _apiKey = Environment.GetEnvironmentVariable("SENDGRID_API_KEY");
             _senderEmail = Environment.GetEnvironmentVariable("SENDGRID_EMAIL");
             _senderName = Environment.GetEnvironmentVariable("SENDGRID_Name");
             _builder = builder;
             _client = new SendGridClient(_apiKey);
+            _configuration = configuration;
+        }
+        public async Task ConfirmRegistration(string email, string firstName)
```

Comment:
Move GetSignUpLetter it to the EmailBuilder and invoke it from UserService. Keep EmailService simple.